store/blockstore.go

```diff
@@ -0,0 +1,124 @@
+package store
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"sync"
+
+	"github.com/lazyledger/optimint/types"
+	"go.uber.org/multierr"
+	"golang.org/x/crypto/sha3"
+)
+
+var (
+	blockPreffix = [1]byte{1}
+	indexPreffix = [1]byte{2}
+)
+
+type DefaultBlockStore struct {
+	db KVStore
+
+	height uint64
+
+	mtx sync.RWMutex
+}
+
+var _ BlockStore = &DefaultBlockStore{}
+
+func NewBlockStore() BlockStore {
+	return &DefaultBlockStore{db: NewInMemoryKVStore()}
+}
+
+func (bs *DefaultBlockStore) Height() uint64 {
+	bs.mtx.RLock()
+	defer bs.mtx.RUnlock()
+	return bs.height
+}
+
+func (bs *DefaultBlockStore) SaveBlock(block *types.Block) error {
+	// TODO(tzdybal): proper serialization & hashing
```

Comment:
Why not make it proper already? 
If we building things from scratch here, sticking with easy and performant solutions like [sha256-simd](https://github.com/minio/sha256-simd) from the beginning makes sense.

Reply:
Goal of this PR is to introduce block store that works. Using sha256-simd actually makes sense as it's drop-in replacement. By this comment I mostly mean that the serialization and hashing code will definitely be modified in the future.